youtube2zim/scraper.py

```diff
@@ -140,6 +151,17 @@ def __init__(
             build_dir=self.build_dir, api_key=self.api_key, cache_dir=self.cache_dir
         )
 
+        # Optimization-cache
+        self.s3_url_with_credentials = s3_url_with_credentials
+        self.use_any_optimized_version = use_any_optimized_version
+        self.video_quality = "low" if self.low_quality else "high"
+        if s3_url_with_credentials:
+            self.s3_storage = KiwixStorage(s3_url_with_credentials)
+            if not self.s3_storage.check_credentials(list_buckets=True):
+                raise AuthenticationError(
```

Comment:
Why use this Exception? Since its failed most probably from user provided input, use `ValueError`.